### ISSUE TITLE
Invalid uri and contens value of filterResponse method for https website

### ISSUE THREAD

**Issue by NozomiIto** [lightbody/browsermob-proxy#238]
Hello.  I'm trying 2.1 beta SNAPSHOT  now.

The ResponseFilter works well for http website,
but does not work well for https website in my environment.

- contents argument is null for https website
- host information is not included in `originalRequest.getUri()` result for https website

The code is like below:

```
public void test() {
        BrowserMobProxy proxy = new BrowserMobProxyServer();
        proxy.start();
        proxy.addResponseFilter(new ResponseFilter() {

            @Override
            public void filterResponse(
                    HttpResponse response,
                    HttpMessageContents contents, HttpRequest originalRequest) {
                if (contents == null && originalRequest != null) {
                    System.out.println("contents is null for " + originalRequest.getUri());
                } else if (contents.getContentType().contains("html")) {
                    System.out.println("HTML contents for " + originalRequest.getUri());                    
                }
            }
        });
        Proxy seProxy = ClientUtil.createSeleniumProxy(proxy);
        DesiredCapabilities capabilities = new DesiredCapabilities();
        capabilities.setCapability(CapabilityType.PROXY, seProxy);
        WebDriver driver = new FirefoxDriver(capabilities);
        driver.get("https://github.com/lightbody/browsermob-proxy");
        driver.quit();
}
```

The standard output is only 1 line:

```
contents is null for /lightbody/browsermob-proxy
```

pom.xml:

```
<dependency>
  <groupId>net.lightbody.bmp</groupId>
  <artifactId>browsermob-core-littleproxy</artifactId>
  <version>2.1.0-beta-1-SNAPSHOT</version>
</dependency> 
<dependency>
  <groupId>org.seleniumhq.selenium</groupId>
  <artifactId>selenium-java</artifactId>
  <version>2.45.0</version>
</dependency>
```

Enviroment:

```
OS X Yosemite: 10.10.3
Firefox: 38.0.1
```

I read [SSL Support](https://github.com/lightbody/browsermob-proxy#ssl-support), but don't do any CA set up. I must do some manual CA set up even for FirefoxDriver?

**Comment by jekh** [comment]
Thanks for creating this issue. There are actually two separate problems you're experiencing:

The first problem, where `contents` is null for responses on HTTPS connections, was caused by a defect in LittleProxy. I've fixed it and released a new BMP version of LP. Pull the latest code from master and rebuild, and this should be fixed.

The second problem you're experiencing, `originalRequest.getUri()` not returning the hostname for HTTPS requests, is works-as-designed. Generally, HTTP clients don't send host information in HTTPS requests, since they are operating over a persistent connection to the host. (Netty simply populates the URI based on the information given it by the HTTP client.)

The easiest way to access the host information for an HTTPS request in a filter is probably to look at the Host header:
```java
String host = HttpHeaders.getHost(request);
```

**Comment by jekh** [comment]
Hi @NozomiIto - I just opened PR #243 after thinking further about your issue. That PR should allow you to retrieve the full URL of the original request more easily. It does create a breaking change to the filters, but it should, by design, be the last breaking change to the filters.

Let me know if you have any thoughts about that PR. I'll hold off on merging it for a bit.

**Comment by jekh** [comment]
Since I believe this has been fixed by #243, I'm going to close the issue. @NozomiIto - feel free to reopen if you're still experiencing issues getting the URL.
